lib/msf/core/payload_generator.rb

```diff
@@ -332,6 +332,11 @@ def generate_payload
         cli_print "Payload size: #{encoded_payload.length} bytes"
         gen_payload = format_payload(encoded_payload)
       end
+
```

Comment:
this needs to move into the elsif block below or you will get a nil exception if a payload could not be generated